src/core/public/analytics/analytics_service.test.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { firstValueFrom, Observable } from 'rxjs';
+import { analyticsClientMock } from './analytics_service.test.mocks';
+import { coreMock, injectedMetadataServiceMock } from '../mocks';
+import { AnalyticsService } from './analytics_service';
+
+describe('AnalyticsService', () => {
+  let analyticsService: AnalyticsService;
+  beforeEach(() => {
+    jest.clearAllMocks();
+    analyticsService = new AnalyticsService(coreMock.createCoreContext());
+  });
+  test('should register some context providers on creation', async () => {
+    expect(analyticsClientMock.registerContextProvider).toHaveBeenCalledTimes(3);
+    await expect(
+      firstValueFrom(analyticsClientMock.registerContextProvider.mock.calls[0][0].context$)
+    ).resolves.toMatchInlineSnapshot(`
+      Object {
+        "branch": "branch",
+        "buildNum": 100,
+        "buildSha": "buildSha",
+        "isDev": true,
+        "isDistributable": false,
+        "version": "version",
+      }
+    `);
+    await expect(
+      firstValueFrom(analyticsClientMock.registerContextProvider.mock.calls[1][0].context$)
+    ).resolves.toEqual({ session_id: expect.any(String) });
+    await expect(
+      firstValueFrom(analyticsClientMock.registerContextProvider.mock.calls[2][0].context$)
+    ).resolves.toEqual({
+      preferred_language: 'en-US',
+      preferred_languages: ['en-US', 'en'],
+      user_agent: expect.any(String),
+    });
+  });
+
+  test('setup should expose all the register APIs, reportEvent and opt-in', () => {
+    const injectedMetadata = injectedMetadataServiceMock.createSetupContract();
+    expect(analyticsService.setup({ injectedMetadata })).toStrictEqual({
+      registerShipper: expect.any(Function),
+      registerContextProvider: expect.any(Function),
+      removeContextProvider: expect.any(Function),
+      registerEventType: expect.any(Function),
+      reportEvent: expect.any(Function),
+      optIn: expect.any(Function),
+      telemetryCounter$: expect.any(Observable),
+    });
+  });
+
+  test('setup should register the elasticsearch info context provider (undefined)', async () => {
+    const injectedMetadata = injectedMetadataServiceMock.createSetupContract();
+    analyticsService.setup({ injectedMetadata });
+    await expect(
+      firstValueFrom(analyticsClientMock.registerContextProvider.mock.calls[3][0].context$)
+    ).resolves.toMatchInlineSnapshot(`undefined`);
```

Comment:
NIT are backquotes necessary here?

Reply:
They are caused by the `toMatchInlineSnapshot`. We could also use `toBeUndefined()`, `toEqual(undefined)`, `toBe(undefined)`

![ron-ogden-so-many-options](https://user-images.githubusercontent.com/5469006/166914855-9846242f-ac5f-4c9b-a14a-361545422c64.gif)